widget/controllers/widget.item.controller.js

```diff
@@ -250,14 +250,45 @@
               return console.error('There was a problem redeeming the coupon');
             };
             UserData.insert(WidgetItem.redeemedItem.data, TAG_NAMES.COUPON_REDEEMED).then(successItem, errorItem);
-
-          }else{
+          } else if (WidgetItem.currentLoggedInUser && WidgetItem.data.settings.toggleEmployeeCode == 'on'){
+           
+            setTimeout(() => {
+              Buildfire.spinner.hide()
+            }, 500);
+            ViewStack.push({
+              template: 'Code',
+              params: {
+                controller: "WidgetCodeCtrl as WidgetCode",
+                item: item,
+                WidgetItem: WidgetItem
+              },
+            });
+          }
+          else{
             buildfire.auth.login({}, function () {
-
             });
           }
         };
 
+        WidgetItem.confirmPasscode = function (passcode) {
+          passcode = angular.element('#passcode').val()
+          currentView = ViewStack.getCurrentView()
+          WidgetItem.item = currentView.params.item
+          if(passcode == WidgetItem.data.settings.employeeCode){
+              WidgetItem.redeemCoupon(currentView.params.item, true);
+              setTimeout(() => {
+                ViewStack.popAllViews();
```

Comment:
shouldn't we do this once redeemcoupon is executed (maybe through a callback?)

Reply:
Moved to successItem callback, so it will only be triggered when item is successfully redeemed